src/components/Panels/Panel.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <section class="panel">
+    <div class="panel__header">
+      <slot name="before-title" v-bind="scope">
+        <template v-if="showBack">
+          <m-icon-button
+            icon="pi-lg pi-arrow-left-s-line"
+            class="panel__back"
+            width="32px"
+            height="32px"
+            flat
+            @click="close"
+          />
+        </template>
+      </slot>
+      <div class="panel__title">
+        <slot name="title" v-bind="scope" />
+      </div>
+      <slot name="after-title" v-bind="scope">
+        <m-icon-button
+          icon="pi-lg pi-close-line"
+          class="panel__close"
+          width="32px"
+          height="32px"
+          flat
+          @click="exit"
+        />
+      </slot>
+    </div>
+
+    <div class="panel__content">
+      <slot />
+    </div>
+
+    <template v-if="slots.footer">
+      <div class="panel__footer">
+        <slot name="footer" v-bind="scope" />
+      </div>
+    </template>
+
+    <template v-if="slots.actions">
+      <div class="panel__actions">
+        <slot name="actions" v-bind="scope" />
+      </div>
+    </template>
+  </section>
+</template>
+
+<script lang="ts" setup>
+  import { computed, useSlots } from 'vue'
+  import { queue } from '@/plugins/Panel'
+
+  const emit = defineEmits<{
+    (event: 'close' | 'exit'): void,
+  }>()
+
+  const slots = useSlots()
+
+  const showBack = computed(() => queue.length > 1)
+
+  const close = (): void => emit('close')
+  const exit = (): void => emit('exit')
+  const scope = { close, exit, showBack }
+</script>
+
+<style lang="scss" scoped>
+.panel {
+  --padding: var(--p-2);
+
+  position: fixed;
+  z-index: 20;
```

Comment:
What's the reasoning for this z-index? I'm not sure the correct value but I think we want to make sure things like toasts, tooltips, and menus will show up above this

Reply:
There's actually not supposed to be a z-index on this component so I'm removing it. The `PanelContainer` component sets the z-index for all panels and it has a custom property so it can be overridden by the app using it. I'm hoping to create a z-layer "map" that sets a custom property for each "layer" we have.

Reply:
Fully support that idea, we're a little all over the place with z-layers right now